`build-extensions` fails on pip install because it can't find `FindASL.cmake`

## Summary
In a fresh Python 3.9.9 environment, I ran `pip install pyomo`, then `pyomo build-extensions`, and get a failure to build the PyNumero libraries as `ampl_function_demo/src/FindASL.cmake` is not found. It doesn't appear to be anywhere in `site_packages`.

### Steps to reproduce the issue

```console
$ pip install pyomo
$ pyomo build-extensions
```

### Error Message

```
**** Building PyNumero libraries ****
-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:18 (INCLUDE):
  INCLUDE could not find load file:

    /home/robert/idaes/fresh-env/lib/python3.9/site-packages/pyomo/contrib/pynumero/src/../../ampl_function_demo/src/FindASL.cmake


-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Configuring incomplete, errors occurred!
See also "/tmp/tmpmu0rmdiu/CMakeFiles/CMakeOutput.log".
ERROR: DistutilsExecError: command '/usr/bin/cmake' failed with exit code 1
```

### Information on your system

Pyomo version: 6.3.0
Python version: 3.9.9
Operating system: Ubuntu 18.04
How Pyomo was installed (PyPI, conda, source): pip

--- a/setup.py
+++ b/setup.py
@@ -247,6 +247,7 @@ setup_kwargs = dict(
     },
     packages = find_packages(exclude=("scripts",)),
     package_data = {
+        "pyomo.contrib.ampl_function_demo": ["src/*"],
         "pyomo.contrib.appsi.cmodel": ["src/*"],
         "pyomo.contrib.mcpp": ["*.cpp"],
         "pyomo.contrib.pynumero": ['src/*', 'src/tests/*'],

